packages/fxa-auth-server/lib/routes/utils/third-party-events.ts

```diff
@@ -57,33 +75,84 @@ async function revokeThirdPartySessions(
 }
 
 // See events at https://developers.google.com/identity/protocols/risc#supported_event_types
-export const eventHandlers = {
+export const googleEventHandlers = {
   'https://schemas.openid.net/secevent/risc/event-type/verification':
-    handleTestEvent,
+    handleGoogleTestEvent,
   'https://schemas.openid.net/secevent/risc/event-type/sessions-revoked':
-    handleSessionsRevokedEvent,
+    handleGoogleSessionsRevokedEvent,
   'https://schemas.openid.net/secevent/risc/event-type/account-disabled':
-    handleAccountDisabledEvent,
+    handleGoogleAccountDisabledEvent,
   'https://schemas.openid.net/secevent/risc/event-type/account-enabled':
-    handleAccountEnabledEvent,
+    handleGoogleAccountEnabledEvent,
   'https://schemas.openid.net/secevent/oauth/event-type/tokens-revoked':
-    handleSessionsRevokedEvent,
+    handleGoogleSessionsRevokedEvent,
 
   // Since we don't store refresh tokens, we can't revoke it. Instead,
   // lets handle this like a sessions-revoked event.
   'https://schemas.openid.net/secevent/oauth/event-type/token-revoked':
-    handleSessionsRevokedEvent,
+    handleGoogleSessionsRevokedEvent,
 
   // We don't support purging accounts, so lets handle this like an account-disabled
   // event, ie revoke all third party sessions and unlink the account.
   'https://schemas.openid.net/secevent/risc/event-type/account-purged':
-    handleAccountDisabledEvent,
+    handleGoogleAccountDisabledEvent,
 
   'https://schemas.openid.net/secevent/risc/event-type/account-credential-change-required':
-    handleSessionsRevokedEvent,
+    handleGoogleSessionsRevokedEvent,
+};
+
+// See https://developer.apple.com/documentation/sign_in_with_apple/processing_changes_for_sign_in_with_apple_accounts
+export const appleEventHandlers = {
```

Comment:
It turns out that Apple does not conform to the RISC standard so I had to modify the code a bit to support this.  I probably over engineered it in the original Google PR, but maybe that is ok.